src/blockdata/script.rs

```diff
@@ -1075,9 +1076,14 @@ impl Encodable for Script {
 }
 
 impl Decodable for Script {
+    #[inline]
+    fn consensus_decode_from_finite_reader<D: io::Read>(d: D) -> Result<Self, encode::Error> {
+        Ok(Script(Decodable::consensus_decode_from_finite_reader(d)?))
+    }
+
     #[inline]
     fn consensus_decode<D: io::Read>(d: D) -> Result<Self, encode::Error> {
-        Ok(Script(Decodable::consensus_decode(d)?))
+        Self::consensus_decode_from_finite_reader(d.take(MAX_VEC_SIZE as u64))
     }
```

Comment:
In ae28a501e0e41a78afb6b01dcf1fba8a689262dd:

Curious why this isn't just a default impl on the trait? This code looks like it's repeated for every individual type.

Reply:
Ah, you already have a default impl going the other way.

FWIW I think it would be ok to have both methods default-impl'd to call each other, with doccomments instructing trait implementors to implement one. I believe that std does this somewhere, though I don't remember where..

Reply:
I think I like it. I'll give it a try.

Reply:
Pushed